backend-shared/config/db/util/application_info_cache.go

```diff
@@ -0,0 +1,411 @@
+	} else {
+		delete(cacheAppState, req.createOrUpdateAppStateObject.Applicationstate_application_id)
+		delete(cacheApp, req.createOrUpdateAppStateObject.Applicationstate_application_id)
+	}
+
+	res, exists := cacheAppState[appState.Applicationstate_application_id]
+	if !exists {
+		// If it's not in the cache, then get it from the database
+		valueFromCache = false
+		err = dbQueries.GetApplicationStateById(req.ctx, &appState)
+		if err != nil {
+			appState = db.ApplicationState{}
+		}
+
+		// Update the cache if we get a result from the database
+		// Update valueFromCache to false, since data is retrieved from db
+		if err == nil && appState.Applicationstate_application_id != "" {
+			if entryInCache, exists := cacheAppState[appState.Applicationstate_application_id]; !exists {
+				entryInCache.appState = appState
+				cacheAppState[appState.Applicationstate_application_id] = entryInCache
+			}
+		}
+
+	} else {
+		// If it is in the cache, return it from the cache
+		valueFromCache = true
+		appState = res.appState
+	}
+
+	req.responseChannel <- applicationInfoCacheResponse{
+		applicationState: appState,
+		valueFromCache:   valueFromCache,
+		err:              err,
+	}
+
+}
+
+func processGetAppMessage(dbQueries db.DatabaseQueries, req applicationInfoCacheRequest, cacheApp map[string]applicationCacheEntry, cacheAppState map[string]applicationStateCacheEntry) {
+
+	app := db.Application{
+		Application_id: req.primaryKey,
+	}
+
+	var err error
+	var valueFromCache bool
+
+	if db.IsEmpty(req.primaryKey) {
+		log.Error("PrimaryKey should not be nil: " + req.primaryKey + " not found")
+	} else {
+		delete(cacheApp, req.createOrUpdateAppStateObject.Applicationstate_application_id)
+		delete(cacheAppState, req.createOrUpdateAppStateObject.Applicationstate_application_id)
+	}
+
+	if res, exists := cacheApp[app.Application_id]; !exists {
+		// If it's not in the cache, then get it from the database
+		valueFromCache = false
+		err = dbQueries.GetApplicationById(req.ctx, &app)
+		if err != nil {
+			app = db.Application{}
+		}
+
+		// Update the cache if we get a result from the database
+		// Update valueFromCache to false, since data is retrieved from db
+		if err == nil && app.Application_id != "" {
+			if entryInCache, exists := cacheApp[app.Application_id]; !exists {
+				entryInCache.app = app
+				cacheApp[app.Application_id] = entryInCache
+			}
+		}
+
+	} else {
+		// If it is in the cache, return it from the cache
+		valueFromCache = true
+		app = res.app
+	}
+
+	req.responseChannel <- applicationInfoCacheResponse{
+		application:    app,
+		valueFromCache: valueFromCache,
+		err:            err,
+	}
+
+}
+
+func processExpireCacheEntriesMessage(cacheApp map[string]applicationCacheEntry, cacheAppState map[string]applicationStateCacheEntry, inputChan chan applicationInfoCacheRequest) {
+
+	for key, elements := range cacheApp {
+		if time.Duration(elements.cacheExpireTime.Minute()) > time.Minute {
```

Comment:
Awesome, it will same one extra check, and it works!
![image](https://user-images.githubusercontent.com/44315191/166684808-46ceb694-bb34-456e-b924-a7a2b3ad13d4.png)